infra-btpsa/usecase.json

```diff
@@ -0,0 +1,86 @@
+{
+    "$schema": "https://raw.githubusercontent.com/SAP-samples/btp-setup-automator/main/libs/btpsa-usecase.json",
+    "aboutThisUseCase": {
+        "name": "SAP Cloud SDK Quickstart",
+        "description": "This usecase provides all necessary information to create and deploy a SAP Cloud SDK based Node.js app a SAP BTP account to Cloud Foundry.",
+        "author": "christian.lechner@sap.com",
+        "testStatus": "tested successfully",
+        "usageStatus": "READY TO BE USED",
+        "relatedLinks": [
+            "https://github.com/Azure-Samples/app-service-javascript-sap-cloud-sdk-quickstart"
+        ]
+    },
+    "services": [
+        {
+            "name": "cloudfoundry",
+            "plan": "free",
+            "category": "ENVIRONMENT"
+        },
+        {
+            "name": "xsuaa",
+            "plan": "application",
+            "category": "SERVICE"
+        },
+        {
+            "name": "destination",
+            "plan": "lite",
+            "category": "SERVICE",
+            "parameters": {
+                "HTML5Runtime_enabled": "true",
+                "init_data": {
+                    "subaccount": {
+                        "existing_destinations_policy": "fail",
+                        "destinations": [
+                            {
+                                "Name": "S4HANACloudSandbox",
+                                "Description": "SAP S/4HANA Cloud Sandbox",
+                                "Type": "HTTP",
+                                "URL": "https://sandbox.api.sap.com/s4hanacloud",
+                                "Authentication": "BasicAuthentication",
+                                "ProxyType": "Internet",
+                                "User": "ODATA_USERNAME",
+                                "Password": "ODATA_USERPWD"
+                            }
+                        ]
+                    }
+                }
+            }
+        },
+        {
+            "category": "SERVICE",
+            "name": "privatelink",
+            "plan": "standard",
+            "entitleonly": true
+        }
+    ],
+    "executeBeforeAccountSetup": [
+        {
+            "description": "Delete quickstart folder in case it already exists (e.g. from a previous script call)",
+            "command": "rm -rf quickstart/"
+        }
+    ],
+    "executeAfterAccountSetup": [
+        {
+            "description": "Get the quickstart code",
+            "command": "git clone https://github.com/Azure-Samples/app-service-javascript-sap-cloud-sdk-quickstart /home/user/quickstart"
```

Comment:
How about referencing the project on the codespace by default instead of an additional clone`?

use instead:
/workspaces/app-service-javascript-sap-cloud-sdk-quickstart

Reply:
> Few findings:
> 
> * Make SAP Private Link optional to support deployment to BTP trial?
> * Adjust CF runtime plan to MEMORY to support trial? "free" seems to be unavailable.
> 
> What if devs are not allowed to create subaccounts but only deploy into it? I was able to use existings names have the btpsa work with existing artifact. Maybe a hint on the docs to re-use existing subaccounts?

In general the `btpsa` does not support SAP BTP Trial landscape due to several limitations of the Trial landscape starting with the regions. Recommendation is to use at least the free tier of SAP BTP.

There is no functionality in `btpsa` to make a service optional. I would therefore keep the entitlement

I enhanced the documentation ion how to deal with existing services (see commit <https://github.com/Azure-Samples/app-service-javascript-sap-cloud-sdk-quickstart/pull/41/commits/c716826411c84cfa60f16e348c7bbb515179d17d>)

Reply:
Good point. This is introduced with commit <https://github.com/Azure-Samples/app-service-javascript-sap-cloud-sdk-quickstart/pull/41/commits/986c65f5d0d127dd96b829a0c6919084083fa255>